src/transformers/trainer.py

```diff
@@ -2109,6 +2117,7 @@ def train(
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
 
         if resume_from_checkpoint is not None:
+            add_safe_globals()
```

Comment:
While this fixes the test failures described in #34631, there are few things to discuss for potentially better implementation:
* I have added allowed list globally. The better approach might be to use context manager like this:
```
with torch.serialization.safe_globals([np.core.multiarray._reconstruct, np.ndarray, np.dtype, np.dtypes.UInt32DType]):
    checkpoint_rng_state = torch.load(rng_file)
```
* Using context manager might require changes in other places in Huggingface which I did not identify yet, need to run more tests.
* I guess I did not allowlisted all what might be needed... This makes me think that we might need a change in Transformers API to allow end users to specify additional list items especially if context manager will be used.
* Another option is to set safe globals on a higher level. In a sense of failing tests described in #34631 this would be in test itself. I personally think this might not be the best approach - for example, tests will need to import `numpy` to allow list its objects. At the moment usage of `numpy` is hidden inside Transformers implementation.
* And yet another option is to ask PyTorch to allowlist certain things. In the case of identified failures - some numpy objects.

@muellerzr, @SunMarc : please, let me know your opinion.

Reply:
@albanD : can you, please, comment whether it makes sense to add any of `numpy` symbols into Pytorch allowed list for safe_globals? I would submit issue / PR on PyTorch side if you believe this makes sense.

Reply:
I agree with `using a context manager`.

In the long term, it might be better if we are able to avoid using `torch.load(weights_only=False)`. But guess we still need to maintain BC.

Reply:
Switched to context manager.